shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+                ((ViewGroup) launchView.getParent()).removeView(launchView);
+                launchView = null;
+              }
+            });
+
+        flutterView.removeFirstFrameListener(this);
+      }
+    });
+
+    // Resets the activity theme from the one containing the launch screen in the window
+    // background to a blank one since the launch screen is now in a view in front of the
+    // FlutterView.
+    //
+    // We can make this configurable if users want it.
+    // TODO: get rid of Activity references.
+    getActivity().setTheme(android.R.style.Theme_Black_NoTitleBar);
+
+    return view;
+  }
+
+  private boolean isSplashScreenDesired() {
+    return getArguments().getBoolean(ARG_IS_SPLASH_SCREEN_DESIRED, false);
+  }
+
+  /**
+   * Extracts a {@link Drawable} from the parent activity's {@code windowBackground}.
+   *
+   * {@code android:windowBackground} is specifically reused instead of other attributes
+   * because the Android framework can display it fast enough when launching the app as opposed
+   * to anything defined in the Activity subclass.
+   *
+   * TODO(mattcarroll): speed aside, should developers be given the opportunity to use different Drawables?
+   *
+   * Returns null if no {@code windowBackground} is set for the activity.
+   */
+  @SuppressWarnings("deprecation")
+  @Nullable
+  private Drawable getSplashScreenDrawableFromActivityTheme() {
+    TypedValue typedValue = new TypedValue();
+    if (!getContextCompat().getTheme().resolveAttribute(
+        android.R.attr.windowBackground,
+        typedValue,
+        true)) {
+      return null;
+    }
+    if (typedValue.resourceId == 0) {
+      return null;
+    }
+    try {
+      return getResources().getDrawable(typedValue.resourceId);
+    } catch (Resources.NotFoundException e) {
+      Log.e(TAG, "Referenced launch screen windowBackground resource does not exist");
+      return null;
+    }
+  }
+
+  /**
+   * Returns the Flutter view used by this {@code Fragment}; will be null before
+   * {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)} is invoked. Will be
+   * non-null after {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)} is invoked, up
+   * until {@link #onDestroyView()} is invoked.
+   */
+  @SuppressWarnings("unused")
+  @Nullable
+  public FlutterView getFlutterView() {
+    return flutterView;
+  }
+
+  /**
+   * Starts running Dart within the FlutterView for the first time.
+   *
+   * The difference between the first run and subsequent reloads is that when FlutterView is
+   * run for the first time, it attempts to reuse an existing Dart isolate.
+   */
+  private void doInitialFlutterViewRun() {
+    runFlutterView(getInitialRoute(), getAppBundlePath(), true);
+  }
+
+  /**
+   * Reloads Dart code with that at the given {@code appBundlePath}, sets Flutter's initial route
+   * to {@code initialRoute}, and then reloads the Dart VM.
+   *
+   * TODO(mattcarroll): is the above description accurate?
```

Comment:
This should probably be resolved in this review :-). @jason-simmons

Reply:
I don't know if we actually support relaunching the engine through this API (see for example https://github.com/flutter/flutter/issues/16998)

@chinmaygarde Should the API offered to engine embedders expose this?  Or would it be preferable for embedders to destroy the engine and create a new instance in order to load a new bundle?